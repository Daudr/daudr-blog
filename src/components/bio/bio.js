import React from 'react'
import { StaticQuery, graphql } from 'gatsby'
import Img from 'gatsby-image'

import { rhythm } from '../../utils/typography'
import Paper from '@material-ui/core/Paper'

export const Bio = ({ isIT = false }) => {
  return (
    <StaticQuery
      query={bioQuery}
      render={data => {
        const { author, social } = data.site.siteMetadata

        const bio = (
          <div>
            Written by <strong>{author}</strong> a freelance web developer that
            works in Italy and in the world. <br />
            His main area of expertise are <strong>Angular</strong>,{' '}
            <strong>React</strong>, <strong>Firebase</strong> &{' '}
            <strong>Serverless Architectures</strong>
            <div style={{ marginTop: rhythm(1 / 2) }}></div>
            You can also find him on
          </div>
        )
        const itBio = (
          <div>
            Scritto da <strong>{author}</strong> uno sviluppatore{' '}
            <strong>Web</strong> che lavora in <strong>Italia</strong> e nel{' '}
            <strong>Mondo</strong> <br />I suoi punti di forza sono{' '}
            <strong>Angular</strong>, <strong>React</strong>,{' '}
            <strong>Firebase</strong> &{' '}
            <strong>Serverless Architectures</strong> e non solo.
            <div style={{ marginTop: rhythm(1 / 2) }}></div>
            Puoi trovarlo anche su
          </div>
        )

        return (
          <Paper
            style={{
              display: `flex`,
              marginTop: rhythm(1.25),
              marginBottom: rhythm(1 / 2),
              padding: rhythm(1 / 2),
            }}
          >
            <Img
              fixed={data.avatar.childImageSharp.fixed}
              alt={author}
              style={{
                marginRight: rhythm(1 / 2),
                marginBottom: 0,
                minWidth: 50,
                borderRadius: `100%`,
              }}
              imgStyle={{
                borderRadius: `50%`,
              }}
            />
            <div>
              {isIT ? itBio : bio}
              {social.map((social, i) => {
                let fixed

                switch (i) {
                  case 0: {
                    fixed = data.facebook.childImageSharp.fixed
                    break
                  }
                  case 1: {
                    fixed = data.linkedin.childImageSharp.fixed
                    break
                  }
                  case 2: {
                    fixed = data.github.childImageSharp.fixed
                    break
                  }
                  case 3: {
                    fixed = data.instagram.childImageSharp.fixed
                    break
                  }
                  case 4: {
                    fixed = data.twitter.childImageSharp.fixed
                    break
                  }
                  case 5: {
                    fixed = data.twitch.childImageSharp.fixed
                    break
                  }
                  default: {
                    break
                  }
                }

                return (
                  <a
                    key={`social-${social.social}`}
                    href={social.link}
                    rel='noopener noreferrer'
                    target='_blank'
                    style={{
                      boxShadow: `none`,
                      color: `currentColor`,
                    }}
                  >
                    <Img
                      fixed={fixed}
                      alt={`${author} ${social.social} link`}
                      style={{
                        marginRight: rhythm(1 / 2),
                        marginBottom: 0,
                        minWidth: 30,
                        maxWidth: `50px`,
                        maxHeight: `50px`,
                      }}
                    />
                  </a>
                )
              })}
            </div>
          </Paper>
        )
      }}
    />
  )
}

const bioQuery = graphql`
  query BioQuery {
    avatar: file(absolutePath: { regex: "/profile-pic.jpg/" }) {
      childImageSharp {
        fixed(width: 50, height: 50) {
          ...GatsbyImageSharpFixed
        }
      }
    }
    facebook: file(absolutePath: { regex: "/facebook.png/" }) {
      childImageSharp {
        fixed(width: 30, height: 30) {
          ...GatsbyImageSharpFixed
        }
      }
    }
    twitter: file(absolutePath: { regex: "/twitter.png/" }) {
      childImageSharp {
        fixed(width: 30, height: 30) {
          ...GatsbyImageSharpFixed
        }
      }
    }
    github: file(absolutePath: { regex: "/github-logo.png/" }) {
      childImageSharp {
        fixed(width: 30, height: 30) {
          ...GatsbyImageSharpFixed
        }
      }
    }
    instagram: file(absolutePath: { regex: "/instagram.png/" }) {
      childImageSharp {
        fixed(width: 30, height: 30) {
          ...GatsbyImageSharpFixed
        }
      }
    }
    linkedin: file(absolutePath: { regex: "/linkedin.png/" }) {
      childImageSharp {
        fixed(width: 30, height: 30) {
          ...GatsbyImageSharpFixed
        }
      }
    }
    twitch: file(absolutePath: { regex: "/twitch.png/" }) {
      childImageSharp {
        fixed(width: 30, height: 30) {
          ...GatsbyImageSharpFixed
        }
      }
    }
    site {
      siteMetadata {
        author
        social {
          social
          link
        }
      }
    }
  }
`

export default Bio
