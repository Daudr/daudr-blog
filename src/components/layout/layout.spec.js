import React from 'react'
import { StaticQuery } from 'gatsby'
import renderer from 'react-test-renderer'

import Layout from './layout'

import { location } from '../../../__mocks__/location.mock'

import { data } from '../../../__mocks__/bio.mock'

beforeEach(() => {
	StaticQuery.mockImplementationOnce(({ render }) => render(data))
})

describe('Layout', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(
				<Layout
					location={location}
					title={`Layout Test`}
					defaultLang='IT'
					setSelectedLanguage={() => {}}
				/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
