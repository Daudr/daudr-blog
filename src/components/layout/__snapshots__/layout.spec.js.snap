// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  style={
    Object {
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": "42rem",
      "padding": "2.625rem 1.3125rem",
    }
  }
>
  <header
    style={
      Object {
        "color": "white",
      }
    }
  >
    <div
      className="background-sky hero"
    />
    <div
      className="flag-select"
      style={
        Object {
          "position": "fixed",
          "right": "10px",
        }
      }
    >
      <div
        className="flag-select "
      >
        <button
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-labelledby="select_flag_button"
          className="flag-select__btn"
          disabled={false}
          id="select_flag_button"
          onClick={[Function]}
          onKeyUp={[Function]}
          style={
            Object {
              "fontSize": "16px",
            }
          }
          type="button"
        >
          <span
            className="flag-select__option flag-select__option--placeholder"
          >
            
          </span>
        </button>
      </div>
    </div>
    <h1
      className="header-title"
      style={
        Object {
          "fontSize": "3.95285rem",
          "lineHeight": "4.375rem",
          "marginBottom": "2.625rem",
          "marginTop": 0,
        }
      }
    >
      <a
        href="/it"
        style={
          Object {
            "boxShadow": "none",
            "color": "inherit",
            "textDecoration": "none",
          }
        }
      >
        Layout Test
      </a>
    </h1>
  </header>
  <main />
  <footer>
    <a
      href="https://www.iubenda.com/privacy-policy/49067529"
      rel="noopener noreferrer"
      style={
        Object {
          "boxShadow": "none",
          "color": "currentColor",
        }
      }
      target="_blank"
      title="Privacy Policy"
    >
      Privacy Policy
    </a>
    <br />
    © 
    2020
    , Michele Da Rin Fioretto - Dauðr
  </footer>
  <div
    className="cookieConsent "
    style={
      Object {
        "alignItems": "baseline",
        "background": "#353535",
        "bottom": "0",
        "color": "white",
        "display": "flex",
        "flexWrap": "wrap",
        "justifyContent": "space-between",
        "left": "0",
        "position": "fixed",
        "width": "100%",
        "zIndex": "999",
      }
    }
  >
    <div
      className=""
      style={
        Object {
          "flex": "1 0 300px",
          "margin": "15px",
        }
      }
    >
      This website uses cookies to enhance the user experience.
    </div>
    <button
      className=""
      id=""
      onClick={[Function]}
      style={
        Object {
          "background": "#ffd42d",
          "border": "0",
          "borderRadius": "0px",
          "boxShadow": "none",
          "color": "black",
          "cursor": "pointer",
          "flex": "0 0 auto",
          "margin": "15px",
          "padding": "5px 10px",
        }
      }
    >
      I undestand
    </button>
  </div>
</div>
`;
